service/src/main/java/bio/terra/policy/db/PaoDao.java

```diff
@@ -74,6 +77,42 @@ public PaoDao(TpsDatabaseConfiguration tpsDatabaseConfiguration) {
     this.tpsJdbcTemplate = new NamedParameterJdbcTemplate(tpsDatabaseConfiguration.getDataSource());
   }
 
+  @Retryable(interceptor = "transactionRetryInterceptor")
+  @Transactional(
+      isolation = Isolation.SERIALIZABLE,
+      propagation = Propagation.REQUIRED,
+      transactionManager = "tpsTransactionManager")
+  public void clonePao(UUID sourceObjectId, UUID destinationObjectId) {
+    DbPao sourcePao = getDbPao(sourceObjectId);
+
+    final String sql =
+        "INSERT INTO policy_object (object_id, component, object_type, sources,"
+            + " attribute_set_id, effective_set_id, predecessor_id)"
+            + " VALUES (:object_id, :component, :object_type, string_to_array(:sources, ','),"
+            + " :attribute_set_id, :effective_set_id, :predecessor_id)";
+
+    final String sourceIds = String.join(",", sourcePao.sources());
+
+    // First we store the policy object with a "forward reference" to the set.
```

Comment:
nit: Is there a second part of this? If not, update comment

Reply:
That's a cut'n'paste from create and is probably wrong in both places now! :)

Reply:
it was and it was :)